plugins/main/vlan/vlan.go

```diff
@@ -60,10 +62,29 @@ func loadConf(bytes []byte) (*NetConf, string, error) {
 
 func createVlan(conf *NetConf, ifName string, netns ns.NetNS) (*current.Interface, error) {
 	vlan := &current.Interface{}
+	var curNetns ns.NetNS
+	var m netlink.Link
+	var err error
+
+	if conf.MasterInContainer != false {
+		curNetns = netns
+	} else {
+		curNetns, err = ns.GetCurrentNS()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get init netns: %v", err)
+		}
+	}
+
```

Comment:
I feel like the code could be tightened up here, but also note that the "RandomVethName()" function must be called in the namespace that veth will be created in, so it must also be under curNetns.Do().  So the first half of the function from LinkByName(conf.Master) until LinkAdd(v) should be within a curNetns.Do() block.